src/test/java/com/exasol/errorcodecrawlermavenplugin/ExasolErrorCrawlerIT.java

```diff
@@ -0,0 +1,92 @@
+     * drop it by hand.
+     */
+    private Path projectDir;
+    private Path projectsSrc;
+
+    @BeforeAll
+    static void beforeAll() throws VerificationException, IOException {
+        final File testDir = ResourceExtractor.simpleExtractResources(ExasolErrorCrawlerIT.class, "/testProject");
+        final Verifier verifier = new Verifier(testDir.getAbsolutePath());
+        verifier.setCliOptions(List.of(//
+                "-Dfile=" + PLUGIN_JAR.toAbsolutePath().toString(), //
+                "-DlocalRepositoryPath=" + mavenRepo.toAbsolutePath(), //
+                "-DpomFile=" + ERROR_CODE_CRAWLER_POM.toAbsolutePath()));
+        verifier.executeGoal("install:install-file");
+        verifier.verifyErrorFreeLog();
+    }
+
+    @BeforeEach
+    void beforeEach() throws IOException {
+        this.projectDir = Files.createTempDirectory("mavenRepo");
+        this.projectsSrc = this.projectDir
+                .resolve(Path.of("src", "main", "java", "com", "exasol", "errorcodecrawlermavenplugin", "examples"));
+        if (!this.projectsSrc.toFile().mkdirs()) {
+            throw new IllegalStateException("Failed to create test projects src folder.");
+        }
+        Files.copy(Objects.requireNonNull(getClass().getClassLoader().getResourceAsStream("testProject/pom.xml")),
+                this.projectDir.resolve("pom.xml"), StandardCopyOption.REPLACE_EXISTING);
+    }
+
+    @AfterEach
+    void afterEach() throws IOException {
+        FileUtils.deleteDirectory(this.projectDir.toFile());
+    }
+
+    @Test
+    void testValidCrawling() throws VerificationException, IOException {
+        Files.copy(EXAMPLES_PATH.resolve("Test1.java"), this.projectsSrc.resolve("Test1.java"),
+                StandardCopyOption.REPLACE_EXISTING);
+        final Verifier verifier = getVerifier();
+        verifier.executeGoal("error-code-crawler:verify");
+        verifier.verifyErrorFreeLog();
+    }
+
+    @Test
+    void testDuplicateErrorCode() throws VerificationException, IOException {
+        Files.copy(EXAMPLES_PATH.resolve("DuplicateErrorCode.java"),
+                this.projectsSrc.resolve("DuplicateErrorCode.java"), StandardCopyOption.REPLACE_EXISTING);
+        final Verifier verifier = getVerifier();
+        final VerificationException exception = assertThrows(VerificationException.class,
+                () -> verifier.executeGoal("error-code-crawler:verify"));
+        assertThat(exception.getMessage(), containsString(
+                "[ERROR] E-ECM-4: Found duplicate error code: 'E-TEST-1' was declared multiple times: DuplicateErrorCode.java:8, DuplicateErrorCode.java:12."));
+    }
+
+    private Verifier getVerifier() throws VerificationException {
+        final Verifier verifier = new Verifier(this.projectDir.toFile().getAbsolutePath());
+        verifier.setLocalRepo(mavenRepo.toAbsolutePath().toString());
+        // verifier.setDebug(true);
```

Comment:
why is this commented out? should we remove it?

Reply:
If you add it you must connect with a debugger to localhost:8000. Otherwise, the test waits for ever. But when debugging it's so helpful that I wanted to have a quick switch.